debugview/src/debug/kotlin/com/revenuecat/purchases/debugview/InternalDebugRevenueCatScreen.kt

```diff
@@ -10,39 +12,66 @@ import androidx.compose.material.Text
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.collectAsState
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
 import androidx.lifecycle.viewmodel.compose.viewModel
+import com.revenuecat.purchases.Package
+import com.revenuecat.purchases.PurchasesTransactionException
 import com.revenuecat.purchases.debugview.models.InternalDebugRevenueCatScreenViewModel
+import com.revenuecat.purchases.debugview.models.InternalDebugRevenueCatScreenViewModelFactory
 import com.revenuecat.purchases.debugview.models.SettingGroupState
 import com.revenuecat.purchases.debugview.models.SettingScreenState
 import com.revenuecat.purchases.debugview.models.SettingState
 import com.revenuecat.purchases.debugview.settings.SettingGroup
+import com.revenuecat.purchases.models.StoreProduct
+import com.revenuecat.purchases.models.StoreTransaction
+import com.revenuecat.purchases.models.SubscriptionOption
 import kotlinx.coroutines.flow.MutableStateFlow
 
 @Composable
 internal fun InternalDebugRevenueCatScreen(
-    screenViewModel: DebugRevenueCatViewModel = viewModel<InternalDebugRevenueCatScreenViewModel>(),
+    onPurchaseCompleted: (StoreTransaction) -> Unit,
+    onPurchaseErrored: (PurchasesTransactionException) -> Unit,
+    screenViewModel: DebugRevenueCatViewModel = viewModel<InternalDebugRevenueCatScreenViewModel>(
+        factory = InternalDebugRevenueCatScreenViewModelFactory(onPurchaseCompleted, onPurchaseErrored),
```

Comment:
I could listen to purchase completion/errors in the composable... But felt that it would be better to move that logic to the viewModel. In order to pass the callbacks as parameters to the view model, we need to create a factory like this.